test_conformance/compiler/test_feature_macro.cpp

```diff
@@ -0,0 +1,750 @@
+//
+// Copyright (c) 2020 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "testBase.h"
+#include <vector>
+#include <algorithm>
+#include "errorHelpers.h"
+
+const char* macro_supported_source = R"(kernel void enabled(global int * buf) {
+        int n = get_global_id(0);
+        buf[n] = 0;
+        #ifndef %s
+            ERROR;
```

Comment:
The point of `ERROR;` is to cause a compiler failure?  Using an error directive seems more explicit.

```suggestion
            #error Expected feature macro was not defined
```